### PR TITLE
[RFC] feat(cron): add an extension to support cron jobs for LoopBack 4

### PR DESCRIPTION
**Pull request by raymondfeng**
This PR adds an extension for LoopBack 4 to provide `cron` capability with https://github.com/kelektiv/node-cron.

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [x] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

**Comment by emonddr** [comment]
Great. Some folks were asking for something like this: https://github.com/strongloop/loopback-next/pull/4896

File: extensions/cron/README.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+### Register Cron jobs
+
+We can create an instance of `CronJob` and bind it to the application context so
+it can be managed by the `CronComponent` life cycle.
+
+```ts
+import {CronJob, asCronJob} from '@loopback/cron';
+
+const job = new CronJob({
+  cronTime: '*/1 * * * * *', // Every one second
```
Comment (emonddr) [inline extensions/cron/README.md]: > // Every one second 

// Every second

File: extensions/cron/README.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  }
+}
+
+const jobBinding = createBindingFromClass(CronJobProvider);
+app.add(jobBinding);
+const now = new Date();
+now.setMilliseconds(now.getMilliseconds() + 10);
+const jobConfig: CronJobConfig = {
+  name: 'job-B', // Name the job
+  cronTime: sendAt(now), // Start the job in 10 ms
```
Comment (emonddr) [inline extensions/cron/README.md]: I love the concept of running cron jobs from a loopback application.
Can we have another example besides one that executes every 10 seconds. Perhaps, register a job to occur at midnight everyday with a name like 'produce report' or something along those lines? Just a thought.

File: extensions/cron/README.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+  runOnInit: false,
+};
+app.configure(jobBinding.key).to(jobConfig);
+```
+
+By default, all cron jobs will be started and stopped by `CronComponent` when
+the application is started or stopped.
+
+To start a cron job after application starts, we can either set `runOnInit` to
```
Comment (emonddr) [inline extensions/cron/README.md]: Are we really `starting` the cron job as soon as application starts? e.g. running the job. Or do we mean `register a cron job with the operating system` as soon as the app starts?

Comment (emonddr) [reply extensions/cron/README.md]: Is the module we are using actually going to enter a cron job into linux OS? Or does it `mimic` the idea of `cron` on linux?

Comment (raymondfeng) [reply extensions/cron/README.md]: It mimics the `cron` scheduler.

File: extensions/cron/README.md (hunk 2 above)
Comment (raymondfeng) [reply extensions/cron/README.md]: The job is for testing purpose and we can not wait for midnight for a CI run. :-).

**Pull request merged**